include/chemfiles/Topology.hpp

```diff
@@ -255,6 +255,18 @@ class CHFL_EXPORT Topology final {
         return residues_;
     }
 
+    /// Assigns topology for all residues in the topology.
+    ///
+    /// The connectivity is assigned based on the maps in pdb_connectivity.hpp,
+    /// you may edit this file for your own means, but by default this table
+    /// contains the 20 'natural' L-Amino acids and the nucleic acids commonly
+    /// found in DNA and RNA (dA, T, dG, dC, A, U, G, C).
+    ///
+    /// @throw Error if an atom name is not in the table.
+    ///
+    /// @example{tests/doc/frame/link_standard_residue_bonds.cpp}
+    void link_standard_residue_bonds();
```

Comment:
I don't know if adding a public methods is the best solution here. The standard residue connectivity only concern PDB related formats, right ? (MMTF, PDB, mmcif) Then, why not have it as an helper function for these formats, and automatically add these bonds?

Reply:
I thought about that as that was my original solution, but decided it might more helpful to allow the software built around Chemfiles to decide if this is necessary.  It allows one to skip this step if necessary or use it for other means like peptide building.

Reply:
But then the software in question must know about this function and decide when to call it. Concerning peptide building, I would imagine that people/software using chemfiles would add the bonds "manually" nevertheless.

I prefer to keep the API as simple as possible (partly because this makes chemfiles easier to use, and partly because more functions means more work, as they have to be mirrored to the C API and all the other languages bindings).

Do you see any other use case that would justify adding a new public function?

Reply:
That's true, but I see this function as somewhat of an analog to Frame::guess_topology (I don't need position data though, so it's in topology), in the sense that it assigns the topology of a residue automatically.

In the software I am developing (that uses Chemfiles for parsing), the logic flow is like this:

Decide if we need to do bond assignment, if so - try to use the standard connectivity and check again. If all else fails, use guess topology. I'm not using the CAPI, and you shouldn't let my one use case dictate decisions, but this is what is guiding my decisions.

Reply:
I would still prefer this functionality to be part of the PDB/MMTF parsing. That way, you can still get good(-ish) topology to work with in your code (and call guess_topology if it is needed).

If there is more demand on exposing this functionality (we could open an issue to discuss this and ask feedback), it should be easy to extract the code into a separated method in the future.

---

Sorry for being picky here, I try to balance functionality, usefulness to existing users and ease of learning and usage for new users, and I dont' always see every possible use case. Please tell me if I'm getting to annoying!

Reply:
It's not required for MMTF parsing as this format lists bonds explicitly. Thus, it's only required for PDB (and CIF?). I can move it to the PDB format sometime this week in any case.

I'm contributing to YOUR project :smile:, so it's your decision in the end.

Reply:
Well I created the project, but I would love to see it grow enough to no longer be MY project :smiley: